cpp/extract_normdata_translations.cc

```diff
@@ -0,0 +1,225 @@
+               *ixtheo_pos = *ixtheo_pos + "_fra";
+        }
+    }
+}
+
+
+void ExtractTranslations(File* const marc_norm_input, 
+                         const std::string german_term_field_spec,
+                         const std::string translation_field_spec,
+                         std::map<std::string,std::string> term_to_translation_map[]) {
+
+    std::set<std::string> german_tags_and_subfield_codes;
+    if (unlikely(StringUtil::Split(german_term_field_spec, ':', &german_tags_and_subfield_codes) < 1))
+        Error("ExtractTranslations: Need at least one translation field");
+
+    std::set<std::string> translation_tags_and_subfield_codes;
+    if (unlikely(StringUtil::Split(translation_field_spec, ':', &translation_tags_and_subfield_codes) < 1))
+        Error("ExtractTranslations: Need at least one translation field");
+
+    unsigned count(0);
+    
+    while (const MarcUtil::Record record = MarcUtil::Record::XmlFactory(marc_norm_input)) {
+
+        std::vector<std::string> german_term;
+
+        // Determine the german term we will have translations for
+
+        for (auto tag_and_subfields : german_tags_and_subfield_codes) {
+             const std::string tag(tag_and_subfields.substr(0, 3));
+             const std::string subfields(tag_and_subfields.substr(3));
+             
+             std::vector<std::string> german_term_for_one_field;
+             record.extractSubfields(tag, subfields, &german_term_for_one_field);
+             
+             // We may get the german term from only one field
+             if (german_term.size() > 1)  
+                 Warning("We have german terms in more than one field for PPN: " + record.getFields()[0]);
+
+             if (not german_term_for_one_field.empty()) 
+                 german_term = german_term_for_one_field;
+        }        
+
+        std::vector<std::string> all_translations;
+
+        // Extract all additional translations
+
+        for (auto tag_and_subfields : translation_tags_and_subfield_codes) {
+            const std::string tag(tag_and_subfields.substr(0, 3));
+            const std::string subfields(tag_and_subfields.substr(3));
+
+            std::vector<std::string> translations;
+            record.extractSubfields(tag, subfields, &translations);
+
+            // For IxTheo-Translations add the language code in the same field
+            augment_ixtheo_tag_with_language(record, tag, translations);
+            
+            all_translations.insert(std::end(all_translations), std::begin(translations), std::end(translations));
+        }
+ 
+        for (auto it = all_translations.begin(); it != all_translations.end(); it++) {
+            if (*it == "IxTheo_eng") {
+                term_to_translation_map[en].emplace(StringUtil::Join(german_term, ' '), *(it+1));
+                ++it;
+            }
+            else if (*it == "IxTheo_fra") {
+                term_to_translation_map[fr].emplace(StringUtil::Join(german_term, ' '), *(it+1));
```

Comment:
Why do we stick a German term in the French entry?  Also, please put single spaces around operators like "+".